### ISSUE TITLE
Feature Request: Tables and images in rich text fields

### ISSUE THREAD

**Issue by chraebsli** [ecamp/ecamp3#4731]
Bisher gibt es nur einige Formatierungsoptionen für Texte wie beim Erstellen von Blöcken (fett, kursiv, durchstrichen, unterstrichen, Listen mit Einrückungen). Zudem kann man zB fett bereits wie bei Markdown verwenden. (`**Text**` ⇒ **Text**)

Während des Planens von Blöcken habe ich gemerkt, dass folgende Formatierungsoptionen noch praktisch wären:
- Tabellen
- Bilder (zB Grafik für ein Spielfeld)
- Überschriften/ Headings
- Markierungen (`<mark>`)

Zudem fände ich es schön, wenn man die Formatierung auch wie bei Markdown wieder löschen könnte (`**` entfernen über Editor statt Popup mit Formatierungsoptionen).

---

Da ich mit [Obsidian](https://obsidian.md/) arbeite, halte ich zudem das [Live Preview Feature](https://youtu.be/NVewTkcnt4s?t=48) sehr nützlich. Vielleicht könnte man dieses in Zukunft auch in dieser Art einbauen.

**Comment by carlobeltrame** [comment]
Thanks for the input. This is definitely a topic worth discussing.
For programmers, markdown is a very familiar and easy-to-use format. For our usual needs, it seems to provide "everything" needed in some way or another.
For general users, markdown is not a familiar way of formatting things. General users usually work with WYSIWYG editors, especially with Word. In Word, it is not common to use special characters to trigger formatting rules.

Personally, I think eCamp v3 should be primarily aimed at general users, not at programmers. As such, formatting text in the button-based Word style should be the main priority. If we manage to support some markdown rules for power users, without getting in the way of (or confusing) normal users which just want to e.g. type a line of four asterisks ****, that is fine with me. But it is much more important to me that e.g. copy and paste from a Word document into eCamp and vice versa yields useful results, than supporting full Markdown. We already support some markdown rules (e.g. ** will yield bold text, and you can toggle bold with Ctrl+B, so keyboard accessibility should not be an issue I think). The obsidian live preview feature which you linked on the other hand, I would assume general users would find it confusing (or at least unfamiliar) to have the asterisks appear and disappear when you navigate around the text.

Technically, we are currently storing the rich text field contents as HTML in the database, so for full markdown support we would either need a transformation in the frontend, or a (probably quite complicated) migration in the backend.

As another technical note, support for tables is currently blocked by print layout 2 not supporting them, although there are multiple ways forward which might be possible in the future. Images are another hard subject, because of storage requirements and image formats. But I take it the discussion in this issue is just about general native support for a markdown text editor, not about the specific rich text features which we support.

**Comment by chraebsli** [comment]
> But it is much more important to me that e.g. copy and paste from a Word document into eCamp and vice versa yields useful results, than supporting full Markdown.

> we are currently storing the rich text field contents as HTML in the database


Yes, of course! I just thought when you store it in Markdown, then why don't use full Markdown. But if you store it as HTML, it doesn't need all the Markdown features. 

---


> support for tables is currently blocked by print layout 2

> Images are another hard subject

Thanks, I'm excited to use them in the future!

**Comment by manuelmeister** [comment]
## Core Meeting Decision

All these features are low prioritized.

- Tables are currently not possible and complicated, we love to do this one day. 
- Images are complicated as well, with all the storage and embedding. 
- Headings are already used and are a accessibility hurdle, because of the different context.
- Marks we did not see the use for this, as we already support italic, bold, underlined. 

We leave this issue open, for future reference.
